gordo_client/io.py

```diff
@@ -81,7 +89,7 @@ def _handle_response(resp: requests.Response, resource_name: Optional[str] = Non
         if _is_json_response(resp):
             return resp.json()
         elif _is_azure_login_response(resp):
-            raise BadGordoRequest(f"Azure login page response found while fetching resource: {resource_name}.")
+            raise BadGordoResponse(f"Azure login page response found while fetching resource: {resource_name}.")
```

Comment:
I would argue that it is still a bad request. That is, all 4xx HTTP codes indicate that there was something wrong with the request performed by the client.

Also, would it be good to have a dedicated subclass of `BadGordoRequest` to indicate that there was an authorization error?

Reply:
from gordo we can get json or model(x-tar). So other responses should raise `BadGordoResponse`